src/pkg/client/cache/oci.go

```diff
@@ -31,6 +32,12 @@ func OciTemp() string {
 func OciTempImage(sum, name string) string {
 	updateCacheSubdir(filepath.Join(OciTempDir, sum))
 
+	name = strings.Replace(strings.TrimPrefix(name, "//"), "/", "_", 3)
```

Comment:
Again, this belongs in a separate commit & PR.

The images aren't being named for uniqueness, but rather for user readability. The unique identifier is the SHA name of the directory which the image lives in, and the name of the image should be in the most readable form possible. I've already committed a patch locally which I'll be submitting as a PR